### PR TITLE
--help should work when $DISPLAY is not set

### PR DESCRIPTION
**Pull request by equalsraf**
see #707

No code yet, just a test case.

### CODE REVIEW

File: src/gui/main.cpp [hunk 1]
```diff
@@ -51,9 +51,10 @@ int cli_main(int argc, char **argv)
 		argsNoFork << argv[i];
 	}
 
-	NeovimQt::App app(argc, argv);
-
-	app.checkArgumentsMayTerminate();
+	QCoreApplication app(argc, argv);
+	QCommandLineParser p;
+	NeovimQt::App::processCommandlineOptions(p, app.arguments());
+	NeovimQt::App::checkArgumentsMayTerminate(p);
```
Comment (equalsraf) [inline src/gui/main.cpp:57]: I''m not entirely happy with this just yet. But i cant fully split cli argument handling from App.

**Comment by equalsraf** [comment]
I had to remove --help-all from the tests,  it was not behaving consistently between CI and my box.

I'm not too happy about the way the code went, but i don't really see an obvious way out. @jgehrig  any bright ideas for that QCoreApplication bit?

**Comment by jgehrig** [comment]
I think this is a bug in Qt.

Any solutions I can think of to work around the issue are ugly. We can't have build a `QApplication` that extends `QCoreApplication` due to the way `Q_OBJECT` works. We also can't use the deprecated `QApplication` CTOR with `GUIenabled = false`.

So far, I can't think of anything better that the code you've written here.

**Comment by equalsraf** [comment]
The only thing that came to mind was to pass the QCommandLineParser as an argument to App (after it is already built and processed arguments)

But I was under the impression that QCommandLineParser was dependent on an existing QCoreApplication instance, but maybe i was wrong?

**Pull request merged**